src/rhsm/connection.py

```diff
@@ -670,19 +673,38 @@ def close_connection(self) -> None:
         Try to close connection to server
         :return: None
         """
-        if self.__conn is not None:
-            # Do proper TLS shutdown handshake (TLS tear down) first
-            if self.__conn.sock is not None:
-                log.debug(f"Closing HTTPS connection {self.__conn.sock}")
-                try:
-                    self.__conn.sock.unwrap()
-                except ssl.SSLError as err:
-                    log.debug(f"Unable to close TLS connection properly: {err}")
-                else:
-                    log.debug("TLS connection closed")
-            # Then it is possible to close TCP connection
+        if self.__conn is None:
+            return
+
+        if self.__conn.sock is None:
+            log.debug(f"Closing TCP connection to {self.__conn.host}")
             self.__conn.close()
-        self.__conn = None
+            self.__conn = None
+            return
+
+        # The server may not send the `close_notify` alert when it closes the
+        # connection on its side.
+        # Here we set a timeout equal to three times the measured response time
+        # to prevent the connection from waiting until TCP timeout.
+        # See RHEL-17345.
+        log.debug(f"Closing TLS connection {self.__conn.sock}")
+
+        response_time: float = self.smoothed_rt or 0.5
```

Comment:
Why `0.5`? Shouldn't we move it to some constant?

Reply:
I am not sure. In theory this can't even happen, not having `smoothed_rt` means we haven't got any connection to close.
If it should be a constant, where? Is it important enough to be in a config file?

Reply:
Sounds good.